app/code/Magento/Inventory/Controller/Adminhtml/Stock/MassDelete.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Controller\Adminhtml\Stock;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\App\Action\Context;
+use Magento\Inventory\Ui\Component\MassAction\Filter;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\InventoryApi\Api\StockRepositoryInterface;
+
+/**
+ * Delete Controller
+ */
+class MassDelete extends Action
+{
+    /**
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = Index::ADMIN_RESOURCE;
+
+    /**
+     * @var StockRepositoryInterface
+     */
+    private $stockRepository;
+
+    /**
+     * @var Filter
+     */
+    private $massActionFilter;
+
+    /**
+     * @param Context $context
+     * @param StockRepositoryInterface $stockRepository
+     * @param Filter $massActionFilter
+     */
+    public function __construct(
+        Context $context,
+        StockRepositoryInterface $stockRepository,
+        Filter $massActionFilter
+    ) {
+        parent::__construct($context);
+        $this->stockRepository = $stockRepository;
+        $this->massActionFilter = $massActionFilter;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute()
+    {
+        if ($this->getRequest()->isPost() !== true) {
+            $this->messageManager->addErrorMessage(__('Wrong request.'));
+
+            return $this->resultRedirectFactory->create()->setPath('*/*');
+        }
+        $deletedItemsCount = 0;
+        foreach ($this->massActionFilter->getIds() as $id) {
+            try {
+                $this->stockRepository->deleteById($id);
+                $deletedItemsCount++;
+            } catch (CouldNotDeleteException $e) {
+                $errorMessage = __('[ID: %1] ', $id) . $e->getMessage();
+                $this->messageManager->addErrorMessage($errorMessage);
+            }
```

Comment:
What about NoSuchEntityException?

Reply:
I think that it is not necessary because deleteById method throw CouldNotDeleteException and this scenario is served. See `Magento\InventoryApi\Api\StockRepositoryInterface`.

Reply:
@bartekszymanski correct